wagtail/admin/views/pages/edit.py

```diff
@@ -161,49 +166,49 @@ def send_commenting_notifications(self):
             send_notification(users, 'updated_comments', {
                 'page': self.page,
                 'editor': self.request.user,
-                'new_comments': [comment for comment in new_comments if comment.pk in threads],
-                'resolved_comments': [comment for comment in resolved_comments if comment.pk in threads],
+                'new_comments': [comment for comment in changes['new_comments'] if comment.pk in threads],
+                'resolved_comments': [comment for comment in changes['resolved_comments'] if comment.pk in threads],
                 'deleted_comments': [],
                 'replied_comments': [
                     {
                         'comment': comment,
                         'replies': replies,
                     }
-                    for comment, replies in replied_comments
+                    for comment, replies in changes['new_replies']
                     if comment.pk in threads
                 ]
             })
 
         return send_notification(global_recipient_users, 'updated_comments', {
             'page': self.page,
             'editor': self.request.user,
-            'new_comments': new_comments,
-            'resolved_comments': resolved_comments,
-            'deleted_comments': deleted_comments,
+            'new_comments': changes['new_comments'],
+            'resolved_comments': changes['resolved_comments'],
+            'deleted_comments': changes['deleted_comments'],
             'replied_comments': [
                 {
                     'comment': comment,
                     'replies': replies,
                 }
-                for comment, replies in replied_comments
+                for comment, replies in changes['new_replies']
             ]
         })
 
-    def log_commenting_changes(self, revision):
+    def log_commenting_changes(self, changes, revision):
         """
         Generates log entries for any changes made to comments or replies.
         """
-        # Skip if this page does not have CommentPanel enabled
-        if 'comments' not in self.form.formsets:
-            return
-
-        new_comments, deleted_comments, resolved_comments, edited_comments, new_replies, deleted_replies, edited_replies = self.get_commenting_changes()
-
         # Skip if no changes were made
-        if not new_comments and not deleted_comments and not resolved_comments and not edited_comments and not new_replies and not deleted_replies and not edited_replies:
+        if (not changes['new_comments']
```

Comment:
This is more explicit, but quite long. If you don't think the explicitness is really needed, we could shorten to a simple `not any(changes.values())`?

Reply:
Thanks! Just realised that this check is completely unnecessary anyway as all the for loops will just skip.